includes/legacy/class-wc-legacy-cart.php

```diff
@@ -29,6 +52,152 @@
 	public $coupon_applied_count = array();
 
 	/**
+	 * Magic getters.
+	 *
+	 * @param string $name Property name.
+	 * @return mixed
+	 */
+	public function __get( $name ) {
+		switch ( $name ) {
+			case 'dp' :
+				return wc_get_price_decimals();
+			case 'prices_include_tax' :
+				return wc_prices_include_tax();
+			case 'round_at_subtotal' :
+				return 'yes' === get_option( 'woocommerce_tax_round_at_subtotal' );
+
+			// map old public props to methods.
+			case 'cart_contents_total' :
+				return $this->get_cart_total( 'raw' );
+			case 'total' :
+				return $this->get_total( 'raw' );
+			case 'subtotal' :
+				return $this->get_subtotal( 'raw' ) + $this->get_subtotal_tax( 'raw' );
+			case 'subtotal_ex_tax' :
+				return $this->get_subtotal( 'raw' );
+			case 'tax_total' :
+				return array_sum( $this->get_merged_taxes( array( 'cart', 'fees' ) ) ); // Should be fees + cart items.
+			case 'taxes' :
+				return $this->get_taxes();
+			case 'shipping_taxes' :
+				return $this->get_shipping_taxes();
+			case 'fee_total' :
+				return $this->get_fee_total( 'raw' );
+			case 'discount_cart' :
+				return $this->get_discount_total( 'raw' );
+			case 'discount_cart_tax' :
+				return $this->get_discount_tax( 'raw' );
+			case 'shipping_total' :
+				return $this->get_shipping_total( 'raw' );
+			case 'shipping_tax_total' :
+				return $this->get_shipping_tax( 'raw' );
+			case 'coupon_discount_amounts' :
+				return $this->get_coupon_discount_totals();
+			case 'coupon_discount_tax_amounts' :
+				return $this->get_coupon_discount_tax_totals();
+			case 'display_totals_ex_tax' :
+			case 'display_cart_ex_tax' :
+				return 'excl' === $this->tax_display_cart;
+			break;
+			case 'cart_contents_weight' :
+				return $this->get_cart_contents_weight();
+			break;
+			case 'cart_contents_count' :
+				return $this->get_cart_contents_count();
+			break;
+			case 'tax' :
+				wc_deprecated_argument( 'WC_Cart->tax', '2.3', 'Use WC_Tax:: directly' );
+				$this->tax = new WC_Tax();
+				return $this->tax;
+			case 'discount_total':
+				wc_deprecated_argument( 'WC_Cart->discount_total', '2.3', 'After tax coupons are no longer supported. For more information see: https://woocommerce.wordpress.com/2014/12/upcoming-coupon-changes-in-woocommerce-2-3/' );
+				return 0;
+			case 'coupons' :
+				return $this->get_coupons();
+		}
+	}
+
+	/**
+	 * Map legacy variables to setters.
+	 *
+	 * @param string $name Property name.
+	 * @param mixed  $value Value to set.
+	 */
+	public function __set( $name, $value ) {
+		switch ( $name ) {
+			case 'cart_contents_total' :
+				$this->set_cart_total( $value );
+			case 'total' :
```

Comment:
We need some 'break's here, otherwise everything under the matched `$name` will also get set to `$value`.